scripts/globals/battlefield.lua

```diff
@@ -68,10 +70,788 @@ xi.battlefield.dropChance =
+    end
+
+    -- Check that players haven't all died or that their dead time is over.
+    local players = battlefield:getPlayers()
+    self:handleWipe(battlefield, players)
+
+    local hasTimeRemaining = xi.battlefield.SendTimePrompts(battlefield, players)
+    if not hasTimeRemaining then
+        for _, player in pairs(players) do
+            player:messageSpecial(zones[player:getZoneID()].text.TIME_IN_THE_BATTLEFIELD_IS_UP)
+        end
+        battlefield:setStatus(xi.battlefield.status.LOST)
+        isExiting = true
+    end
+
+    if isExiting then
+        -- Remaining in battlefield for an extended time
+        if cutsceneTimer > 0 then
+            return
+        end
+
+        if status == xi.battlefield.status.LOST then
+            for _, player in pairs(players) do
+                player:messageSpecial(zones[player:getZoneID()].text.PARTY_MEMBERS_HAVE_FALLEN)
+            end
+        end
+
+        battlefield:cleanup(true)
+    end
+end
+
+function Battlefield:onBattlefieldRegister(player, battlefield)
+end
+
+function Battlefield:onBattlefieldStatusChange(battlefield, players, status)
+    -- Remove battlefield effect for players in alliance not inside battlefield once the battlefield gets locked. Do this only once.
+    if status == xi.battlefield.status.LOCKED and battlefield:getLocalVar("statusRemoval") == 0 then
+        battlefield:setLocalVar("statusRemoval", 1)
+
+        for _, player in pairs(players) do
+            local alliance = player:getAlliance()
+            for _, member in pairs(alliance) do
+                if member:hasStatusEffect(xi.effect.BATTLEFIELD) and not member:getBattlefield() then
+                    member:delStatusEffect(xi.effect.BATTLEFIELD)
+                end
+            end
+        end
+    end
+end
+
+function Battlefield:onBattlefieldEnter(player, battlefield)
+    local initiatorId, _ = battlefield:getInitiator()
+    if player:getID() == initiatorId then
+        if #self.requiredKeyItems > 0 then
+            for _, item in ipairs(self.requiredKeyItems) do
+                player:delKeyItem(item)
+            end
+            if self.requiredKeyItems.message ~= 0 then
+                player:messageSpecial(self.requiredKeyItems.message, unpack(self.requiredKeyItems))
+            end
+        end
+
+        for _, item in ipairs(self.requiredItems) do
+            if item.message ~= 0 then
+                player:messageSpecial(item.message, 0, 0, 0, item[0])
+            end
+        end
+    end
+
+    local ID = zones[self.zoneId]
+    player:messageSpecial(ID.text.ENTERING_THE_BATTLEFIELD_FOR, 0, self.index)
+    if self.maxPlayers > 6 then
+        player:messageSpecial(ID.text.MEMBERS_OF_YOUR_ALLIANCE, 0, 0, 0, self.maxPlayers)
+    elseif self.maxPlayers > 1 then
+        player:messageSpecial(ID.text.MEMBERS_OF_YOUR_PARTY, 0, 0, 0, self.maxPlayers)
+    else
+        -- TODO: Need capture of message to know the proper format
```

Comment:
We should get an issue raised for this TODO to cover the proper case

Reply:
I'll ask my capture buddy to try this. I imagine it happens with Maat fights and other solo battlefields.

Reply:
Cannot find use case so I'm simplifying it for now. I think SE leaves a lot of text hanging around so it might not be used anymore or is used in a very specific use case. https://github.com/LandSandBoat/server/pull/2824/commits/e592f93bcf3dd42153c76583ae14d87cd1c2b556